src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -0,0 +1,698 @@
+    /// @param _keeper Address of the new keeper.
+    function setKeeper(address _keeper) external onlyOwner {
+        address oldKeeper = keeper;
+
+        keeper = _keeper;
+        emit KeeperUpdated(_keeper, oldKeeper);
+    }
+
+    /// @notice Updates the frequency at which rewards are processed by the keeper. Can only be called by owner.
+    /// @param _claimFrequency The new claim frequency in seconds.
+    function setClaimFrequency(uint256 _claimFrequency) external onlyOwner {
+        uint256 oldClaimFrequency = claimFrequency;
+
+        claimFrequency = _claimFrequency;
+        emit ClaimFrequencyUpdated(_claimFrequency, oldClaimFrequency);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Manager - Config
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Updates the current value for the minimum executable price (in bps as proportion of an oracle price)
+    ///         for a CRV to WETH swap. The value should be more than the minimum value. Can be called by the owner or
+    ///         the manager.
+    /// @param _minOutBpsCrvToWeth The new value in bps.
+    function setMinOutBpsCrvToWethVal(uint256 _minOutBpsCrvToWeth) external onlyOwnerOrManager {
+        if (_minOutBpsCrvToWeth > MAX_BPS) {
+            revert InvalidBps(_minOutBpsCrvToWeth);
+        }
+
+        uint256 minOutBpsCrvToWethMin = minOutBpsCrvToWeth.min;
+        if (_minOutBpsCrvToWeth < minOutBpsCrvToWethMin) {
+            revert LessThanBpsMin(_minOutBpsCrvToWeth, minOutBpsCrvToWethMin);
+        }
+
+        uint256 oldMinOutBpsCrvToWethVal = minOutBpsCrvToWeth.val;
+        minOutBpsCrvToWeth.val = _minOutBpsCrvToWeth;
+
+        emit MinOutBpsCrvToWethValUpdated(_minOutBpsCrvToWeth, oldMinOutBpsCrvToWethVal);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Owner
+    ////////////////////////////////////////////////////////////////////////////
+
+    function createPrivateVault(uint256 _pid) external onlyOwner {
+        _createPrivateVault(_pid);
+    }
+
+    /// @notice Takes a given amount of asset from the owner and stakes them on private vault. Can only be called by owner.
+    /// @param _pid Pid target to stake into its appropiate vault
+    /// @param _amountAsset Amount of asset to be lock.
+    function depositInPrivateVault(uint256 _pid, uint256 _amountAsset) external onlyOwner returns (bytes32 kekId) {
+        if (_amountAsset == 0) {
+            revert NothingToDeposit();
+        }
+
+        address vaultAddr = privateVaults[_pid];
+        if (vaultAddr == address(0)) {
+            revert NoPrivateVaultForPid(_pid);
+        }
+
+        IStakingProxy proxy = IStakingProxy(vaultAddr);
+        address stakingToken = proxy.stakingToken();
+        IFraxUnifiedFarm farm = IFraxUnifiedFarm(proxy.stakingAddress());
```

Comment:
Should we cache `stakingToken` and `stakingAddress` to storage?

Reply:
in this case was only used once, not sure if worth caching here

always we can derived those two vars from the proxy itself which is on storage, for the convex-frax infra i decided only to store the keks and proxies